src/dos/dos_locale.cpp

```diff
@@ -0,0 +1,2904 @@
+
+// ***************************************************************************
+// Country info - time/date format, currency, etc.
+// ***************************************************************************
+
+enum class LocalePeriod : uint8_t {
+	// Values for Modern period reflect the KDE/Linux system settings and
+	// currency information from Wikipedia - they won't produce 100% same
+	// result as old MS-DOS systems, but should at least provide reasonably
+	// consistent user experience with certain host operating systems.
+	// Please provide them for each supported country!
+	Modern,
+
+	// Values for Historic period mimic the locale present in (in order of
+	// priority): MS-DOS 6.22, Windows ME, PC-DOS 2000, DR-DOS 7.03,
+	// FreeDOS. Possibly utilize other software, preferrably from before
+	// 2000, or even 1995.
+	// Do not omit it if the definitions are the same as in 'Modern', as
+	// 'Modern' might easily change in the future.
+	// NOTE: If only possible, keep the Euro currency away from Historic!
+	Historic
+};
+
+enum class Separator : uint8_t {
+	Space      = 0x20, // ( )
+	Apostrophe = 0x27, // (')
+	Comma      = 0x2c, // (,)
+	Dash       = 0x2d, // (-)
+	Period     = 0x2e, // (.)
+	Slash      = 0x2f, // (/)
+	Colon      = 0x3a, // (:)
+	Semicolon  = 0x3b, // (;)
+};
+
+struct LocaleInfoEntry {
+	DosDateFormat date_format;
+	Separator     date_separator;
+	DosTimeFormat time_format;
+	Separator     time_separator;
+
+	// Several variants of currency symbol can be provided, ideally start
+	// from "best, but few code pages can display it", to "worst, but many
+	// code pages can handle it".
+	// Add the rarely available symbols even for Historic locales; old code
+	// pages won't display them nevertheless and the engine will select
+	// more common one.
+	// If the currency symbol is written in a non-Latin script, provide also
+	// a Latin version (if there is any); we do not want to display 'ºδ⌠'
+	// where the currency was 'Δρχ' (drachma), we really prefer "Dp"
+	std::vector<std::string> currency_symbols_utf8;
+
+	// Currency codes are selected based on ISO 4217 standard,
+	// they are used as a last resort solution if no UTF-8 encoded symbols
+	// can be losslessly converted to current code page.
+	std::string currency_code;
+
+	uint8_t currency_precision; // digits after decimal point
+	DosCurrencyFormat currenct_format;
+
+	Separator thousands_separator;
+	Separator decimal_separator;
+
+	// Current locale systems (like Unicode CLDR) do not seem to specify
+	// list separators anymore. On 'https://answers.microsoft.com' one can
+	// find a question:
+	// - Why does Excel seem to use ; and , differently per localization?
+	// And the top answer is:
+	// - In countries such as the USA and UK, the comma is used as list
+	//   separator. (...)
+	//   In countries that use comma as decimal separator, such as many
+	//   continental European countries, using a comma both as decimal
```

Comment:
"European countries, except for the UK" :trollface: (it's the truth)

Reply:
This is actually a quotation - reformatted the text.